src/main/java/io/stargate/sgv2/jsonapi/service/operation/model/impl/DeleteOperation.java

```diff
@@ -107,20 +137,67 @@ private QueryOuterClass.Query buildDeleteQuery() {
    * @param queryExecutor
    * @param query
    * @param doc
-   * @return Uni<Boolean> `true` if deleted successfully, else `false`
+   * @param attempt
+   * @return Uni<DeleteResponse>
    */
-  private static Uni<Boolean> deleteDocument(
-      QueryExecutor queryExecutor, QueryOuterClass.Query query, ReadDocument doc) {
-    query = bindDeleteQuery(query, doc);
-    return queryExecutor
-        .executeWrite(query)
+  private Uni<DeleteResponse> deleteDocument(
+      QueryExecutor queryExecutor,
+      QueryOuterClass.Query query,
+      ReadDocument doc,
+      AtomicInteger attempt) {
+    final Uni<ReadDocument> documentToDelete =
+        Uni.createFrom()
+            .item(attempt.get())
+            .onItem()
+            .transformToUni(
+                attemptValue -> {
+                  if (attemptValue > 0) {
+                    return readDocumentAgain(queryExecutor, doc);
+                  } else {
+                    return Uni.createFrom().item(doc);
+                  }
+                });
+    return documentToDelete
         .onItem()
         .transformToUni(
-            result -> {
-              if (result.getRows(0).getValues(0).getBoolean()) {
-                return Uni.createFrom().item(true);
+            docToDelete -> {
+              if (docToDelete == null) {
+                return Uni.createFrom().item(DeleteResponse.MODIFIED_BY_CONCURRENT_TRANSACTION);
               } else {
-                return Uni.createFrom().item(false);
+                QueryOuterClass.Query boundQuery = bindDeleteQuery(query, docToDelete);
+                return queryExecutor
+                    .executeWrite(boundQuery)
+                    .onItem()
+                    .transform(
+                        result -> {
+                          if (result.getRows(0).getValues(0).getBoolean()) {
+                            return DeleteResponse.DELETED;
+                          } else {
+                            return DeleteResponse.CONCURRENCY_FAILURE;
+                          }
+                        });
```

Comment:
lwt is send back as a failure or? I see this here: https://github.com/stargate/stargate/blob/8c1208a461e68e3068fadb576f63ac812d78a1bf/coordinator/bridge/src/main/java/io/stargate/bridge/service/ExceptionHandler.java#L267-L279

Or is this something different?

Reply:
Not sure what this is. But incase of LWT failure it returns false in result as expected.

Reply:
But if there is no document, what would be returned? The same? Then it's OK, next read would fail..

Reply:
Yes the same logic will handle, if no document and we try to delete with txid = 'xxxx' it will return false.
`cassandra@cqlsh:purchase_database> select * from myt where pk = '1';

 pk | col1 | col2
----+------+------

cassandra@cqlsh:purchase_database> delete from myt where pk = '1' if col1 = 'aaa';

 [applied]
-----------
     False
`